### DISCUSSION TITLE
Accessing blogs from custom page

### DISCUSSION THREAD

**Discussion by tobias-tengler** [facebook/docusaurus#4800] [Q&A] [answered]
Hi, I would like to render a little carousel of the most recent blogs on my index page.
I'm using the `@docusaurus/preset-classic` and have only set `showReadingTime: true` in the blog configuration.
I was using `gatsby-transformer-mdx` before, where I could just query for the blogs:
```js
const data = useStaticQuery(graphql`
  query getIndexPageData {
    allMdx(
      limit: 3
      filter: { frontmatter: { path: { glob: "/blog/**/*" } } }
      sort: { fields: [frontmatter___date], order: DESC }
    ) {
      edges {
        node {
          id
          fields {
            readingTime {
              text
            }
          }
          frontmatter {
            featuredImage {
              childImageSharp {
                fluid(maxWidth: 800, pngQuality: 90) {
                  ...GatsbyImageSharpFluid
                }
              }
            }
            path
            title
            date(formatString: "MMMM DD, YYYY")
          }
        }
      }
    }
  }
`);
``` 
Is something like this possible with Docusaurus? Can I access the frontmatter?
I've read about `useGlobalData` and `usePluginData`, but I can't seem to access the blogs through them.
The following gives me an error that the plugin does not exist:
```js
usePluginData("docusaurus-plugin-content-blog");
```

**Comment by slorber** [comment, marked as answer]
We don't support to query plugin data with GraphQL like Gatbsy does.

It is however possible to achieve what you want in various ways:
- create a plugin and use `loadContent` + `contentLoaded` + `addRoute` to create your homepage (equivalent of creating a page with Gatsby node)
- generate a json representing the blog post with a node script as a pre-build step and require it in your index page.
I admit it is not ideal but I don't have many other alternatives.

We also plan to allow you to "extend" an existing content plugin: for example you'll be able to add a blog index page by reusing the internal data that the blog plugin already read on the filesystem. Some discussions here: https://github.com/facebook/docusaurus/issues/4138

If you want to create an index page that involves reading the data of multiple plugins, unfortunately I'm not sure we'll be able to solve this problem easily, as unlike Gatsby, we don't have a unified data cache that you can query in different ways according to your usecase. I'll need to think about this more to see if a solution is possible.
